modules/tensor_mechanics/src/materials/ADLAROMCreepStressUpdate.C

```diff
@@ -0,0 +1,535 @@
+        "immobile_dislocation_density_forcing_function",
+        "Optional forcing function for immobile dislocation. If provided, the immobile dislocation "
+        "density will be reset to the function value at the beginning of the timestep. Used for "
+        "testing purposes only.");
+
+    params.addParam<FunctionName>(
+        "old_creep_strain_forcing_function",
+        "Optional forcing function for the creep strain from the previous timestep. If provided, "
+        "the old creep strain will be reset to the function value at the beginning of the "
+        "timestep. Used for testing purposes only.");
+
+    params.addParam<bool>("verbose", false, "Flag to add verbose output"););
+
+template <ComputeStage compute_stage>
+ADLAROMCreepStressUpdate<compute_stage>::ADLAROMCreepStressUpdate(
+    const InputParameters & parameters)
+  : ADRadialReturnCreepStressUpdateBase<compute_stage>(parameters),
+    _rom(getUserObject<LAROMData>("rom_data")),
+    _temperature(adCoupledValue("temperature")),
+    _environmental(adCoupledValue("environmental_factor")),
+    _window(getParam<Real>("input_window_limit")),
+    _window_failure(
+        parameters.get<MooseEnum>("input_window_failure_action").getEnum<WindowFailure>()),
+    _extrapolate_stress(getParam<bool>("extrapolate_to_zero_stress")),
+    _verbose(getParam<bool>("verbose")),
+    _mobile_dislocations(declareADProperty<Real>(_base_name + "mobile_dislocations")),
+    _mobile_dislocations_old(getMaterialPropertyOld<Real>(_base_name + "mobile_dislocations")),
+    _initial_mobile_dislocations(getParam<Real>("initial_mobile_dislocation_density")),
+    _max_mobile_increment(isParamValid("max_relative_mobile_dislocation_increment")
+                              ? getParam<Real>("max_relative_mobile_dislocation_increment")
+                              : _rom.getMaxRelativeMobileInc()),
+    _mobile_function(isParamValid("mobile_dislocation_density_forcing_function")
+                         ? &getFunction("mobile_dislocation_density_forcing_function")
+                         : NULL),
+    _mobile_dislocation_increment(0.0),
+    _mobile_old(0.0),
+    _immobile_dislocations(declareADProperty<Real>(_base_name + "immobile_dislocations")),
+    _immobile_dislocations_old(getMaterialPropertyOld<Real>(_base_name + "immobile_dislocations")),
+    _initial_immobile_dislocations(getParam<Real>("initial_immobile_dislocation_density")),
+    _max_immobile_increment(isParamValid("max_relative_immobile_dislocation_increment")
+                                ? getParam<Real>("max_relative_immobile_dislocation_increment")
+                                : _rom.getMaxRelativeImmobileInc()),
+    _immobile_function(isParamValid("immobile_dislocation_density_forcing_function")
+                           ? &getFunction("immobile_dislocation_density_forcing_function")
+                           : NULL),
+    _immobile_dislocation_increment(0.0),
+    _immobile_old(0.0),
```

Comment:
This seems dangerous because the value is outside of the window and because a zero dislocation density is pretty nonphysical. Is this setting required for passing to the user object?

Reply:
This is just parameter initialization for the increment container. It is set later.